profiles/permissions.py

```diff
@@ -13,7 +13,7 @@ def has_object_permission(self, request, view, obj):
 
 class SavedCompaniesListPermission(BasePermission):
     def has_permission(self, request, view):
-        if request.query_params.get("filters") == "is_saved":
+        if request.query_params.get("is_saved"):
```

Comment:
Not nice, normally permissions should not have any relations with filtering.

I don't mind going with this option (it will be definitely not the most questionable code in this project). But if you have time and inspiration, I'd consider the next options:

1. Just ignore `is_saved` in permission class (this actually will be right approach, because filtering options should not be a reason to return 403). Of course, you'll have to update logic in this filter (in such way, that when unauthorized user tries to use this filter — empty queryset is returned).
2. Instead of using the same endpoint for 'normal' profiles list, and saved profiles list — create two separate endpoints. The one for saved profiles list will have `IsAuthenticated` in permission classes.

Both options are equally OK (each has own advantages/disadvantages, and in different cases I used both of them in real projects) and both are much better than the current approach.

Reply:
Permission removed and logic in is_saved filter updated.

Thanks for the recommendation!